internal/compose/compose.go

```diff
@@ -275,54 +275,58 @@ func (p *Project) WaitForHealthy(opts CommandOptions) error {
 		return err
 	}
 
+	timeout := time.After(10 * time.Minute)
+	ticker := time.NewTicker(1 * time.Second)
+
 	containerIDs := strings.Split(strings.TrimSpace(b.String()), "\n")
 	for {
 		if signal.SIGINT() {
 			return errors.New("SIGINT: cancel waiting for policy assigned")
 		}
 
-		healthy := true
-
-		logger.Debugf("Wait for healthy containers: %s", strings.Join(containerIDs, ","))
-		descriptions, err := docker.InspectContainers(containerIDs...)
-		if err != nil {
-			return err
-		}
+		select {
```

Comment:
Let me challenge this and consider another approach.

We already have "for" loop with a 1-sec delay per iteration. The function can store `time.Now()` and just compare it if the `startTime.Add(timeout)` is greater than `time.Now`. We don't need to add channels/select/ticker. WDYT?

Reply:
I spent some time searching differences between `time.Sleep` and `time.Tick`, and the only thing I found is that a tick tries to guarantee the interval, while sleep is unaware of execution times. There seems to be no difference in performance for the 2 implementations.

In general I prefer the ticker approach as it's behaviour is explicit, but in this case I don't see particular benefits as guaranteeing the interval is not a concern.

I'll rework it with your approach.

Reply:
I think I prefer the ticker and select approach too. It also works better than time arithmetic when you have multiple source of events, as may happen if/when we have contexts here.

Reply:
But we can wait to have contexts here, not a strong opinion :slightly_smiling_face:

Reply:
I'm a bit confused here as adding a for-select block will introduce another nesting level and make the function hard to read. TBH I would follow the KISS rule, so if it's possible to solve with a single if-clause, I would go for it.

In general, we can consider refactoring of this area, as there is a similar logic in the [SIGINT](https://github.com/elastic/elastic-package/blob/8d9732d0c4364d28ba1c34f2ad7fe91dfc7d2785/internal/signal/sigint.go#L24) function. It would be great, If we can refactor all parts (SIGINT, timeout, waiter logic, contexts). You have my sword :)

Reply:
By running a quick search I found there are 5 `for {` loops in the codebase. I think it would make a lot of sense to generalize the approach, so interval/timeout/signals are handled consistently. 

Are you ok to keep it for a future PR?  I already have a ticker-based `Waiter` struct implementation to use as a baseline, but I'm blocked on other work streams so I would prefer to push this refactor to the back burner for a bit.